app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+    );
+  }
+
+  List<PopupMenuEntry<String>> _buildAppBarActionMenuItems(ActerPin pin) {
+    List<PopupMenuEntry<String>> actions = [];
+
+    //Get my membership details
+    final membership =
+        ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
+    if (membership != null) {
+      //Check for can post pin permission
+      if (membership.canString('CanPostPin')) {
+        //EDIT PIN TITLE MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintTitleDialog(context, ref, pin),
+            child: Text(L10n.of(context).editTitle),
+          ),
+        );
+
+        //EDIT PIN DESCRIPTION MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            key: PinDetailsPage.editBtnKey,
+            onTap: () => showEditPintDescriptionDialog(context, ref, pin),
+            child: Text(L10n.of(context).editDescription),
+          ),
+        );
+      }
+
+      final canRedact = ref.watch(canRedactProvider(pin));
+      if (canRedact.valueOrNull == true) {
+        final roomId = pin.roomIdStr();
+        //DELETE PIN MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showRedactDialog(
+              context: context,
+              pin: pin,
+              roomId: roomId,
+            ),
+            child: Text(
+              L10n.of(context).removePin,
+              style: TextStyle(color: Theme.of(context).colorScheme.error),
+            ),
+          ),
+        );
+      } else {
+        //REPORT PIN MENU ITEM
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showReportDialog(context, pin),
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Atlas.warning_thin,
+                  color: Theme.of(context).colorScheme.error,
+                ),
+                const SizedBox(width: 10),
+                Text(L10n.of(context).reportPin),
+              ],
+            ),
+          ),
+        );
+      }
+    }
+
+    return actions;
+  }
+
+  Widget _buildPinHeaderUI(ActerPin pin) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Row(
+            children: [
+              pinIconUI(),
+              const SizedBox(width: 12),
+              Expanded(
+                child: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    pinTitleUI(pin),
+                    pinSpaceNameUI(pin),
+                  ],
+                ),
+              ),
+            ],
+          ),
+          pinDescriptionUI(pin),
+        ],
+      ),
+    );
+  }
+
+  Widget pinIconUI() {
```

Comment:
In future, this will be dynamic and putted into common widget for Icon Identity management.